woocommerce/woocommerce-ios
# [Custom Amounts M4] Refunds

Closes: #11196


## Description

With this PR we address everything related to custom amounts and refunds:

- Refactor code to use custom amounts instead of fees. We set the line between the view model (we use custom amounts)  and model (we keep using fees), to differentiate between the user-facing wording and the backend, including the model, consistent with the rest of Woo's existing nomenclature.
- Make the custom amounts refundable by default if we only have custom amounts (no items). The reason behind is that if the user wants to refund an order with just a custom amount e.g. Simple Payments, we can assume that they want to refund the custom amount/fee. If they have products and custom amounts, they might want to choose what to refund. Starting with nothing to refund might make it easier for them.

## Testing instructions


We need to prepare three type or **completed** orders:

1. Only with items
2. Only with custom amounts
3. With both items and custom amounts

Go to refund each of these screens. Please check that:

1. When only with items, they're not enabled to refund by default
2. Only with custom amounts, they're enabled to refund by default
3. With both items and custom amounts, nothing is enabled to refund by default

Also, please ensure that the fee wording is being replaced by custom amounts.


## Screenshots


### Only with items

https://github.com/woocommerce/woocommerce-ios/assets/1864060/05ecb92e-f34a-44e2-b29d-512248c7e48e

### Only with custom amounts

https://github.com/woocommerce/woocommerce-ios/assets/1864060/e98c18d4-8ab9-4b5e-b725-d891d29f3f3a

### With both items and custom amounts

https://github.com/woocommerce/woocommerce-ios/assets/1864060/b16e6246-6ca3-465e-a0ca-f0ddbd700c27

---
